appengine-java21/analytics/src/main/webapp/WEB-INF/appengine-web.xml

```diff
@@ -14,9 +14,14 @@
 -->
 <!-- [END_EXCLUDE] -->
 <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
-  <threadsafe>true</threadsafe>
-  <runtime>java8</runtime>
-  <env-variables>
+
+  <runtime>java21</runtime>
+   <app-engine-apis>true</app-engine-apis>
+  <system-properties>
+    <property name="appengine.use.EE8" value="true"/>
```

Comment:
I'm not sure why we are doing this? I'm trying to understand if a new to App Engine user would have this in their sample code or if this would be for someone upgrading and this is meant just for upgrading? It also feels like it's going to create additional areas that need to get updated over time in addition to runtime, now we'll have multiple Enterprise Edition properties?

Reply:
Here, we need 2 changes to move code from java8 to java21 without big or any java code change.
1/ <app-engine-apis>true</app-engine-apis>
and 
2/ <property name="appengine.use.EE8" value="true"/> to avoid using the new Jakarta EE10 name space.
Fully documented on GAE public doc.

Reply:
@ludoch to restate my question, is the purpose of this specific sample only to highlight how to migrate from java 8 to java21 or is this sample also meant to explain to users new to App Engine who want to deploy fresh to a new runtime. As I'm reviewing changes here I have to base my review based on that update. If this is meant for legacy migration path, it might be helpful to document that so that the intent is clear. Right now it looks like these are samples meant to support folks wanting to run on the java 21 runtime. It could be implied that running bundled legacy services is required in that scenario.